vfs.go

```diff
@@ -383,7 +383,7 @@ func (v *VFS) Read(fh *VFSfh, offset uint64, nbytes uint64) ([]byte, error) {
 // appends data at the end of the file. If the file does not exist,
 // it will be created.
 func (v *VFS) Write(fh *VFSfh, bytes []byte) error {
-	cbuffer := C.CBytes(bytes)
+	cbuffer := slicePtr(bytes)
```

Comment:
This looks like a very, very good change: `C.CBytes` [makes a **copy** of the `[]byte` it’s given](https://pkg.go.dev/cmd/cgo#:~:text=//%20Go%20%5B%5Dbyte%20slice%20to%20C%20array%0A//%20The%20C%20array%20is%20allocated%20in%20the%20C%20heap%20using%20malloc.%0A//%20It%20is%20the%20caller%27s%20responsibility%20to%20arrange%20for%20it%20to%20be%0A//%20freed%2C%20such%20as%20by%20calling%20C.free%20(be%20sure%20to%20include%20stdlib.h%0A//%20if%20C.free%20is%20needed).%0Afunc%20C.CBytes(%5B%5Dbyte)%20unsafe.Pointer), which meant that every time we were calling `someVFS.Write(fh, ourData)`, we were _leaking a copy of our data_. Since this is now using `slicePtr`, we’re giving it the address of our existing `[]byte` and telling core to look in there.

Reply:
…although I think we may also want a `defer runtime.KeepAlive(bytes)` right after this, because otherwise `bytes` is never referenced once it enters the `C.tiledb_vfs_write` call and thus would be eligible for GC as it is in progress.

Reply:
Done, thanks!

Reply:
really nice catch